### PR TITLE
Fix Additive Production

### PR DESCRIPTION
**Pull request by hacatu**
See https://github.com/GTNewHorizons/ForestryMC/pull/55

### CODE REVIEW

**Review by Alastors — CHANGES_REQUESTED** [review]
You forgot the BM frames

**Comment by Alastors** [comment]
> You forgot the BM frames

https://github.com/GTNewHorizons/MagicBees/tree/master/src/main/java/magicbees/main/utils/compat/bloodmagic here's the folder with all 3

**Review by Alastors — APPROVED** [review]
Code itself looks fine, though I'm not sure how the 1.8x -> +1.8 translates really, I've been told it's fine, and likewise while I am concerned about the -9001f I've been told it *should* be fine so everything else looks in order

**Review by Runakai1 — APPROVED** [review]

**Review by Runakai1 — CHANGES_REQUESTED** [review]
While I think it's generally fine. Did you test it? I'm kind of unsure since you changed the logic a bit

**Comment by hacatu** [comment]
@Alastors yeah a negative modifier is ok pending the parent PR in forestry, so this should be merged after

@Runakai1 I have been running my forked versions of these forestry mods in my actual world, so I'm very sure the electrical stimulator changes in Binnie and basically all the changes besides the magic apiary change are working (although I have not done like a rigorous statistical test to check the rates are consistent).
I will double check the magic apiary isn't somehow broken

**Comment by hacatu** [comment]
Ok I did a very simple test (run cave dwelling shorter lifespan blinding speed dreaming bee for one cycle) ...

The expected yield is

- Unboosted Soul Combs: `20 bee ticks * (38.8% for first roll + 27.0% for second roll) = 13.2`
- Unboosted Watery Combs: `20 * (26.6% + 18.6%) = 9.0`
- Boosted Soul Combs: `20 * (45.2% + 31.5%) = 15.3`
- Boosted Watery Combs: `20 * (31.0% + 21.7%) = 10.5`

The observed yield was

- Unboosted Soul Combs: 14
- Unboosted Watery Combs: 9
- Boosted Soul Combs: 9
- Boosted Watery Combs: 7

Obviously, this is the opposite of what we would expect, but it actually isn't worrying to me.

For the Boosted scenario the variance is 9.27 for Soul Combs and 7.68 for Watery Combs.

We can then use the normal approximation to find (eg in Python with `from statistics import NormalDist as N`)
- `P(Boosted Soul Combs ≤ 9) = N(15.3, 9.27).cdf(9) ≈ 24.8%`
- `P(Boosted Watery Combs ≤ 7) = N(10.5, 7.68).cdf(7) ≈ 32.4%`

In other words, only testing for 20 bee ticks isn't a large sample size.

I'm convinced that the fix to the magic apiary is correct, but I can re-run the test for a longer time if you want

**Comment by hacatu** [comment]
Darn it, I did the test wrong, `NormalDist` takes standard deviation, not variance, so I should have taken the square roots of 9.27 and 7.68 ... and then we get P values of 1.9% and 10.3%, so I should re-test 😞

**Comment by hacatu** [comment]
Yup there was an issue, the P values don't lie (well 0.2 ish percent of the time a P value of 0.2 ish percent would literally lie ...) but it was in my changes to forestry.  It's fixed now, and I confirmed the `productionModifier` values use debug logging

**Review by kuba6000 — CHANGES_REQUESTED** [review]
Please adjust all frames, not only negative ones.
1.5x -> +0.5
2x -> +1

**Comment by Runakai1** [comment]
> Magic/Resiliant frames were only nerfed from +2 to +1.1 and +1.2 respectively because they require significantly more investment than the base forestry frames and should always have been stronger
> 
> The boosted magic apiary was only nerfed from +1.8 to +1.4 because it requires automated botania mana and breeding any bees to be cave dwelling
> 
> The basic blood frames were not nerfed from +2 because they are expensive and have a moderately high running LP cost (1 bucket per frame per bee tick (27.5 seconds))

Please don't include rebalancing issues in this now. Make another PR if you want to buff and rebalance frames.

**Review by Runakai1 — APPROVED** [review]

**Review by Dream-Master — APPROVED** [review]
